### DISCUSSION TITLE
Filtering data in a data grid

### DISCUSSION THREAD

**Discussion by NikitaGolubovskiy** [AvaloniaUI/Avalonia#11729] [Q&A]
I need to organize data filtering for data grid.
In version 0.10.21, this can be done by accessing the Item property, which is of type DataGridCollectionView.
In version 11.0.0-rc1.1, this field has been replaced with a DataConnection property that has the DataGridConnection type, which contains a CollectionView property of type DataGridCollectionView I need.
The problem is that the new property has an internal scope and I can't access it.
How can I achieve filtering in the new version?

**Comment by timunie** [comment]
I personally like to use DynamicData for filtering stuff in the ViewModel. Here is an example for Xamarin, but it works quite similar for Avalonia 

https://www.xamboy.com/2021/01/25/filtering-and-sorting-with-dynamic-data-in-xamarin-forms-part-2/

**Reply by timunie** [reply]
Note: It's shipped with ReactiveUI if you have that already installed.

**Comment by pjt33** [comment]
For people who find this discussion by searching, as I did, it's possible currently (11.3.4) to use codebehind to set up a filter on the `CollectionView`.

    public MyControl()
    {
        InitializeComponent();

        // Delay until the CollectionView has been created
        myDataGrid.Initialized += (_, __) =>
        {
            myDataGrid.CollectionView.Filter = item => item is MyItemType myItem && ...;
        };
    }

You may also need to set up some event hooks to call `myDataGrid.CollectionView.Refresh()`. No additional libraries required.

**Comment by JakkuSakura** [comment]
https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/pull/382
<img width="1134" height="671" alt="image" src="https://github.com/user-attachments/assets/1766f740-2ceb-41f7-a0aa-5c3f69a1e3d1" />
